src/basket-service/models/basket-item.ts

```diff
@@ -138,14 +145,36 @@ export class BasketItem {
     return this.feeInShopperCurrency ? this.feeInShopperCurrency.value : null;
   }
 
+  getDeliveryFeeAmount () {
+    return this.deliveryFeeInShopperCurrency ? this.deliveryFeeInShopperCurrency.value : null;
+  }
+
+  getOriginalSalePriceAmountWithoutDeliveryFee () {
```

Comment:
Is originalSalePriceAmount() always going to include deliveryFee() and FeeAmount()? What if location doesn't include fees?

Reply:
Yes I was planning to keep originalSalePriceAmount() as it is, which means it will include both delivery fee and service fee. We use this to render the final total price in the order summary in basket_fe today. I created the new methods `getOriginalSalePriceAmountWithoutDeliveryFee ` and `getOriginalSalePriceAmountWithoutFees` to help render the subtotal price in the frontend depending on whether the location includes fees - so we'd still check in the frontend on the basket what the location setting is

Reply:
But this function has no context about whether or not fees are included, correct? If we were to try and get `getOriginalSalePriceAmountWithoutFees` on a NYC show we would actually get the wrong salePrice?

Reply:
Yeah these functions have no context on whether fees are included or not, but I was thinking we can choose the right method to use on the frontend depending on which component we're rendering the price in and the show location. If we use `getOriginalSalePriceAmountWithoutFees` on a NYC show, we'd get the list/sale price with no service fee or delivery fee, to use for the subtotal price

Reply:
Super confused here. Lets use two examples, same setup for both. 
```
1 ticket
Face Value 100
Sale Price 50
Service Fee: 2
Delivery Fee: 5
```
If this was a London Show vs a NYC show, will `getOriginalSalePriceAmountWithoutFees` return different numbers or the same number? I am confused if I call `getOriginalSalePriceAmountWithoutFees` on NYC will I get $50 or $43?

If the returned amount is $50 in both cases I'm totally cool with it, otherwise I'm thinking it makes more sense to either have access to the two fees and let frontend (who knows whether or not to subtract) to do the math. Or give this function the knowledge needed to make the correct decision.

Reply:
ideally no one except platform is doing math. platform should provide all necessary data and clients should be dumb

Reply:
I think we might not be using the same terminology again, which may be causing some of the confusion. 😅 Here is your example reframed using how "sale price" is defined in the sdk: 
```
1 ticket
Face Value 100
List Price 50 (perhaps due to a discount of 50)
Service Fee 2
Delivery Fee 5
Sale Price 57 (= 50 + 2 + 5)
```

In that example above, here's what the different methods would return, regardless of the show location: 
- `getOriginalSalePriceAmount()` = 57
- `getOriginalSalePriceAmountWithoutDeliveryFee()` = 52
- `getOriginalSalePriceAmountWithoutFees()` = 50

In the frontend, we would use whichever method we'd need per the component / show location.

Reply:
> ideally no one except platform is doing math. platform should provide all necessary data and clients should be dumb

Yeah I see how that might be an even better approach. Based on the code that's already there, my thinking was to keep the code changes for this story minimal to get us the accurate numbers, but still improve what was done in basket_fe by bringing the calculations into js-sdk, so it's cleaner in the frontend

Reply:
I think the client is dumb if both are returning the same amounts, just wanted to be clear on that one point. If thats the case I really think it's completely fine.